File: templates/problems/05_binary_search/04_find_min_rotated/python/solution.py
def solve_find_min(nums: list[int]) -> int:
    """Find the minimum in a rotated sorted array.

    Time: O(log n), Space: O(1).
    """
    lo, hi = 0, len(nums) - 1
    while lo < hi:
        mid = lo + (hi - lo) // 2
        if nums[mid] > nums[hi]:
            lo = mid + 1
        else:
            hi = mid
    return nums[lo]

File: templates/problems/05_binary_search/04_find_min_rotated/typescript/solution.ts
export function solveFindMin(nums: number[]): number {
  // Find the minimum in a rotated sorted array.
  // Time: O(log n), Space: O(1).
  let lo = 0;
  let hi = nums.length - 1;
  while (lo < hi) {
    const mid = lo + Math.floor((hi - lo) / 2);
    if (nums[mid] > nums[hi]) {
      lo = mid + 1;
    } else {
      hi = mid;
    }
  }
  return nums[lo];
}
